### ISSUE TITLE
[Accessibility] Text contrast

### ISSUE THREAD

**Issue by Henni** [owncloud/core#24297]
As the topic came up again I want to revive the accessibility discussion regarding our font.
I want to focus on the contrast and set aside the font weight.

The w3c recommends a minimum contrast of 4.5:1 (https://www.w3.org/TR/WCAG/#visual-audio-contrast) while the files app has a contrast of down to 2.46:1.

Comparison:

| current | minimum | enhanced |
| :-: | :-: | :-: |
| ![image](https://cloud.githubusercontent.com/assets/1267827/14832593/547c4294-0bfb-11e6-87e3-24151c95e94b.png) | ![image](https://cloud.githubusercontent.com/assets/1267827/14832621/6d4b12d2-0bfb-11e6-826c-d5c4205e9dce.png) | ![image](https://cloud.githubusercontent.com/assets/1267827/14832644/7ee862ec-0bfb-11e6-8c19-fcd8c93cd625.png) |
| 2.46:1 `#999` | 4.53:1 `#727272` | 7.02:1 `#555555` |

This can be reproduced with [Accessibility Developer Tools for Chrome](https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb?hl=en)

@owncloud/designers @owncloud/accessibility what should we do about this?

**Comment by skjnldsv** [comment]
Minimum is probably better, should we add an option?

**Comment by MorrisJobke** [comment]
> Minimum is probably better, should we add an option?

Why an option. I guess this is about the color scale for the time stamp, because it gets lighter the older a file is and darker the newer a file is, or am I wrong?

**Comment by skjnldsv** [comment]
Because i always think letting the user choose is a good way to improve the user experience! :)
It's not always the case, but it could be a good idea?
I don't know, but I think having an option to activate an "accessibility mode" is an excellent feature to promote owncloud. I worked on several project with those kinds of improvements, and always had a positive feedback. Dyslexic-friendly fonts, font-size, contrast, dark mode... Always good to have! ;)

**Comment by Henni** [comment]
@skjnldsv In my opinion readable text shouldn't be 'optional'. High contrast doesn't only help visually impaired but makes it easier for everyone to grasp the content of a page.

@MorrisJobke The date maybe wasn't the best example for this as its contrast depends on the age of the file. But this problem also applies to the header (2.85:1):
![image](https://cloud.githubusercontent.com/assets/1267827/14845745/975ceef0-0c60-11e6-9287-96fdcc4656dc.png)
and the file size (2.59:1):
![image](https://cloud.githubusercontent.com/assets/1267827/14845772/bd284710-0c60-11e6-845b-9b02b16e6c12.png)

**Comment by MorrisJobke** [comment]
> Because i always think letting the user choose is a good way to improve the user experience! :)
> It's not always the case, but it could be a good idea?

An option is always the worst UX ;) We should have good defaults, that work for everyone and not outsource stuff we don't want to take care of to the user. If we decide on a good default nobody needs to tweak something. Or did you notice a font tweaker on github or facebook ;)

**Comment by skjnldsv** [comment]
You're right, there is not for the fonts only! But facebook has a High contrast mode (HCM). Windows has the function has well. Linux (Gnome, xfce...) :)

**Comment by Henni** [comment]
@skjnldsv This high contrast mode should extend on the default accessibility (i.e. switch from minimum contrast to enhanced contrast or even higher)

**Comment by skjnldsv** [comment]
Probably :)

**Comment by jancborchardt** [comment]
> I don't know, but I think having an option to activate an "accessibility mode" is an excellent feature to promote owncloud.

The best example against this is the iPhone. Enabling »accessibility mode« actually changes nothing at all about the interface. Accessibility should not be an afterthought but baked in. There are not »normal people« and »disabled people«, everyone needs accesibility to a certain extent, be it colorblindness, small monitor, shitty light conditions etc.

> Because i always think letting the user choose is a good way to improve the user experience! :)

Sorry but this is definitely not the case. No one wants to fiddle with settings. :)

To get to the point: @Henni both your examples (date and size) are so low contrast because they are examples of the lower spectrum of the algorithm which changes the shade of grey of the text depending on the size and date. There specifically, the issue never really came up and I would prefer to leave it as it is. Unless of course the people who complain about specifically mean this. Then we could just cap the algorithm at a darker value.

**Comment by Henni** [comment]
@jancborchardt header and file size colors don't depend on this algorithm.
Similar contrast issues can also be found in the sidebar (2.85 and 3.54): 
![image](https://cloud.githubusercontent.com/assets/1267827/14913380/ed095ec6-0e02-11e6-93cb-4babd13319a0.png)

**Comment by jancborchardt** [comment]
Can you do a PR or screenshot of how it would look like with #727272 then? :)

**Comment by Henni** [comment]
![image](https://cloud.githubusercontent.com/assets/1267827/14914229/ad8a5cb8-0e08-11e6-947d-ef77ef4eac96.png)
![image](https://cloud.githubusercontent.com/assets/1267827/14914267/faf0241a-0e08-11e6-8802-bf977a378153.png)

**Comment by jancborchardt** [comment]
I guess it looks good, yeah. And we should adhere with accessibility standards, so good call. Do you want to open a PR for that then? :)

**Comment by Henni** [comment]
@jancborchardt I'll prepare a PR.

**Comment by skjnldsv** [comment]
Got it!

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `status/STALE` is set and there were no updates for 7 days. Feel free to reopen this issue if you deem it appropriate.

(This is an automated comment from [GitMate.io](https://gitmate.io).
